src/module-elasticsuite-core/Index/Mapping/Field.php

```diff
@@ -354,19 +353,26 @@ private function getFieldAnalyzers()
      *
      * @return array
      */
-    private function getPropertyConfig($analyzer = self::ANALYZER_UNTOUCHED)
+    private function getPropertyConfig($analyzer = self::ANALYZER_UNTOUCHED): array
     {
         $fieldMapping = ['type' => $this->getType()];
 
-        if ($this->getType() == self::FIELD_TYPE_TEXT && $analyzer == self::ANALYZER_UNTOUCHED) {
-            $fieldMapping['type']  = self::FIELD_TYPE_KEYWORD;
-        } elseif ($this->getType() == self::FIELD_TYPE_TEXT) {
-            $fieldMapping['analyzer'] = $analyzer;
-            if ($analyzer === self::ANALYZER_SORTABLE) {
-                $fieldMapping['fielddata'] = true;
-            }
-        } elseif ($this->getType() == self::FIELD_TYPE_DATE) {
-            $fieldMapping['format'] = implode('||', $this->dateFormats);
+        switch ($this->getType()) {
+            case self::FIELD_TYPE_TEXT:
+                if ($analyzer === self::ANALYZER_UNTOUCHED) {
+                    $fieldMapping['type'] = self::FIELD_TYPE_KEYWORD;
+                    $fieldMapping['ignore_above'] = 256;
```

Comment:
Use a constant instead of using hardcoded value

Reply:
My mistake. I will correct